### PR TITLE
Chemical toolbox updates

### PR DESCRIPTION
**Pull request by hechth**
This PR includes an update to openbabel convert which allows adding the properties using the ```--append``` flag to the title of whatever is being converted.

I also included a test for it.

### CODE REVIEW

File: chemicaltoolbox/openbabel/ob_convert.xml [hunk 1]
```diff
@@ -132,7 +132,11 @@
         #if float($ph) >= 0:
             -p $ph
         #end if
-
+        
+        #if $appendproperties != 'None':
```
Comment (bgruening) [inline chemicaltoolbox/openbabel/ob_convert.xml:136]: I think it can never be None, its not optional="true"

Comment (hechth) [reply chemicaltoolbox/openbabel/ob_convert.xml:136]: It was none when I tried running without it - I can make it optional if that is preferred.

Comment (bgruening) [reply chemicaltoolbox/openbabel/ob_convert.xml:136]: The question is, should it be optional or not? If so I guess it does not hurt to add optional=true to the mix.

**Comment by bgruening** [comment]
```
 +Linting tool /home/runner/work/galaxytools/galaxytools/chemicaltoolbox/openbabel/ob_convert.xml
Failed linting
.. WARNING (TestsExpectNumOutputs): Test 1: should specify 'expect_num_outputs' if outputs have filters
.. WARNING (TestsExpectNumOutputs): Test 2: should specify 'expect_num_outputs' if outputs have filters
.. WARNING (TestsExpectNumOutputs): Test 3: should specify 'expect_num_outputs' if outputs have filters
.. WARNING (TestsExpectNumOutputs): Test 4: should specify 'expect_num_outputs' if outputs have filters
.. WARNING (TestsExpectNumOutputs): Test 5: should specify 'expect_num_outputs' if outputs have filters
.. WARNING (TestsExpectNumOutputs): Test 6: should specify 'expect_num_outputs' if outputs have filters
.. WARNING (TestsExpectNumOutputs): Test 7: should specify 'expect_num_outputs' if outputs have filters
.. WARNING (TestsExpectNumOutputs): Test 8: should specify 'expect_num_outputs' if outputs have filters
.. CHECK (TestsNoValid): 8 test(s) found.
.. INFO (OutputsNumber): 2 outputs found.
.. INFO (InputsNum): Found 53 input parameters.
.. ERROR (InputsSelectOptionDuplicateText): Select parameter [oformat_opts_selector] has multiple options with the same text content

File: chemicaltoolbox/openbabel/ob_convert.xml (hunk 1 above)
Comment (hechth) [reply chemicaltoolbox/openbabel/ob_convert.xml:136]: it is optional

**Comment by hechth** [comment]
> ```
>  +Linting tool /home/runner/work/galaxytools/galaxytools/chemicaltoolbox/openbabel/ob_convert.xml
> Failed linting
> .. WARNING (TestsExpectNumOutputs): Test 1: should specify 'expect_num_outputs' if outputs have filters
> .. WARNING (TestsExpectNumOutputs): Test 2: should specify 'expect_num_outputs' if outputs have filters
> .. WARNING (TestsExpectNumOutputs): Test 3: should specify 'expect_num_outputs' if outputs have filters
> .. WARNING (TestsExpectNumOutputs): Test 4: should specify 'expect_num_outputs' if outputs have filters
> .. WARNING (TestsExpectNumOutputs): Test 5: should specify 'expect_num_outputs' if outputs have filters
> .. WARNING (TestsExpectNumOutputs): Test 6: should specify 'expect_num_outputs' if outputs have filters
> .. WARNING (TestsExpectNumOutputs): Test 7: should specify 'expect_num_outputs' if outputs have filters
> .. WARNING (TestsExpectNumOutputs): Test 8: should specify 'expect_num_outputs' if outputs have filters
> .. CHECK (TestsNoValid): 8 test(s) found.
> .. INFO (OutputsNumber): 2 outputs found.
> .. INFO (InputsNum): Found 53 input parameters.
> .. ERROR (InputsSelectOptionDuplicateText): Select parameter [oformat_opts_selector] has multiple options with the same text content
> ```

This is also the case for many other tools in here :)

**Comment by hechth** [comment]
@bgruening this should be ready now

**Comment by bgruening** [comment]
> > ```
> >  +Linting tool /home/runner/work/galaxytools/galaxytools/chemicaltoolbox/openbabel/ob_convert.xml
> > Failed linting
> > .. WARNING (TestsExpectNumOutputs): Test 1: should specify 'expect_num_outputs' if outputs have filters
> > .. WARNING (TestsExpectNumOutputs): Test 2: should specify 'expect_num_outputs' if outputs have filters
> > .. WARNING (TestsExpectNumOutputs): Test 3: should specify 'expect_num_outputs' if outputs have filters
> > .. WARNING (TestsExpectNumOutputs): Test 4: should specify 'expect_num_outputs' if outputs have filters
> > .. WARNING (TestsExpectNumOutputs): Test 5: should specify 'expect_num_outputs' if outputs have filters
> > .. WARNING (TestsExpectNumOutputs): Test 6: should specify 'expect_num_outputs' if outputs have filters
> > .. WARNING (TestsExpectNumOutputs): Test 7: should specify 'expect_num_outputs' if outputs have filters
> > .. WARNING (TestsExpectNumOutputs): Test 8: should specify 'expect_num_outputs' if outputs have filters
> > .. CHECK (TestsNoValid): 8 test(s) found.
> > .. INFO (OutputsNumber): 2 outputs found.
> > .. INFO (InputsNum): Found 53 input parameters.
> > .. ERROR (InputsSelectOptionDuplicateText): Select parameter [oformat_opts_selector] has multiple options with the same text content
> > ```
> 
> This is also the case for many other tools in here :)

I know :( maybe I find time to fix it.

**Comment by hechth** [comment]
@bgruening could you trigger the workflow again? I fixed the errors in the documentation and the tests.

**Comment by bgruening** [comment]
Done. Danke das du da rein schaust.

**Comment by hechth** [comment]
So does the CI lint all tools in the repo?

**Comment by bgruening** [comment]
No, only the folder in which you are in.

**Comment by bgruening** [comment]
Outch, I see, you have no touched many folders. It should only lint the folder in which the .shed.yml file is that corresponds to your XML file that you touched. So per subfolder.

**Comment by hechth** [comment]
I initally only made changes to the openbabel stuff - alright, I can discard all changes to files outside that folder

**Comment by hechth** [comment]
Ok so I'll fix whatever I touched so far and then we can merge

**Comment by bgruening** [comment]
The changes look good, but its a lot of work I fear.

File: chemicaltoolbox/xchem-deep/.shed.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: xchem-deep
```
Comment (bgruening) [inline chemicaltoolbox/xchem-deep/.shed.yml:1]:
```suggestion
name: xchem_transfs_scoring
```

File: chemicaltoolbox/fragmenter/fragmenter.xml [hunk 3]
```diff
@@ -1,11 +1,10 @@
-<tool id="fragmenter" name="Fragmenter" version="0.1">
+<tool id="fragmenter" name="Fragmenter" version="0.1+galaxy0">
```
Comment (bgruening) [inline chemicaltoolbox/fragmenter/fragmenter.xml:1]:
```suggestion
<tool id="fragmenter" name="Fragmenter" version="0.1+galaxy0" profiler="23.0">
```

File: chemicaltoolbox/fragmenter/merger.xml [hunk 4]
```diff
@@ -31,19 +30,7 @@
...
-            <output name="outfile" ftype="smi" file='fragmenter_on_CID_2244_recap.smi'/>
-        </test>
-        <test>
-            <param name="infile" ftype="sdf" value="CID_2244.sdf"/>
-            <param name='rules' value='recap' />
-            <param name='oformat' value='smi' />
-            <param name='mark' value='--mark-sticky-ends' />
-            <output name="outfile" ftype="smi" file='fragmenter_on_CID_2244_recap_mark.smi'/>
-        </test>
+        <test expect_failure="true" />
```
Comment (bgruening) [inline chemicaltoolbox/fragmenter/merger.xml:33]: Its not worth the effort, but we should also not push tools that fail, lets remove the .shed.yml and not publish this tool

File: chemicaltoolbox/openbabel/ob_grep.xml [hunk 5]
```diff
@@ -3,8 +3,9 @@
     <!--parallelism method="multi" split_inputs="infile" split_mode="to_size" split_size="10000" shared_inputs="" merge_outputs="outfile"></parallelism-->
     <macros>
         <import>macros.xml</import>
-        <token name="@GALAXY_VERSION@">0</token>
+        <token name="@GALAXY_VERSION@">1</token>
     </macros>
+    <options sanitize="False"/>
```
Comment (bgruening) [inline chemicaltoolbox/openbabel/ob_grep.xml:8]: should this be moved the to actual parameter that needs it? and whitespace does not need to be allowed I guess?

File: chemicaltoolbox/pgchem/pgchem/db_fetch_by_id.xml [hunk 6]
```diff
@@ -1,11 +1,13 @@
-<tool id="pgchem_db_fetch_by_id" name="Fetch" version="0.1">
+<tool id="pgchem_db_fetch_by_id" name="Fetch" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
```
Comment (bgruening) [inline chemicaltoolbox/pgchem/pgchem/db_fetch_by_id.xml:1]:
```suggestion
<tool id="pgchem_db_fetch_by_id" name="Fetch" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@" profile="23.0">
```

File: chemicaltoolbox/pgchem/pgchem/db_filter.xml [hunk 7]
```diff
@@ -1,11 +1,13 @@
-<tool id="pgchem_db_filter" name="Filter" version="0.1">
+<tool id="pgchem_db_filter" name="Filter" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
```
Comment (bgruening) [inline chemicaltoolbox/pgchem/pgchem/db_filter.xml:1]:
```suggestion
<tool id="pgchem_db_filter" name="Filter" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@" profile="23.0">
```

File: chemicaltoolbox/pgchem/pgchem/db_filter.xml [hunk 8]
```diff
@@ -120,15 +112,22 @@
...
+      <when value="DrugLike"/>  
+      <when value="LeadLike"/>  
     </conditional>
   </inputs>
   <outputs>
     <!-- Only fetch smiles from the database since SDF retrieval takes ages -->
     <data format="smi" name="outfile"/>
   </outputs>
+  <tests>
+    <test expect_failure="true" />
```
Comment (bgruening) [inline chemicaltoolbox/pgchem/pgchem/db_filter.xml:129]: lets not do this, rather not push the tool to the TS ... rename the .shed.yml to something else and we need to come back and fix this tool if there is a use-case ...

File: chemicaltoolbox/pgchem/pgchem/db_similarity_search.xml [hunk 9]
```diff
@@ -1,11 +1,13 @@
-<tool id="pgchem_db_sim_search" name="Similarity search" version="0.1">
+<tool id="pgchem_db_sim_search" name="Similarity search" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
```
Comment (bgruening) [inline chemicaltoolbox/pgchem/pgchem/db_similarity_search.xml:1]:
```suggestion
<tool id="pgchem_db_sim_search" name="Similarity search" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@" profile="23.0">
```

File: chemicaltoolbox/pgchem/pgchem/db_substructure_search.xml [hunk 10]
```diff
@@ -1,11 +1,14 @@
-<tool id="pgchem_db_sub_search" name="Substructure search" version="0.1">
+<tool id="pgchem_db_sub_search" name="Substructure search" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
```
Comment (bgruening) [inline chemicaltoolbox/pgchem/pgchem/db_substructure_search.xml:1]:
```suggestion
<tool id="pgchem_db_sub_search" name="Substructure search" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@" profile="23.0">
```

File: chemicaltoolbox/simsearch/simsearch.xml [hunk 11]
```diff
@@ -2,7 +2,7 @@
     <description>of fingerprint data sets with chemfp</description>
     <macros>
         <token name="@TOOL_VERSION@">1.6.1</token>
-        <token name="@GALAXY_VERSION@">0</token>
+        <token name="@GALAXY_VERSION@">2</token>
```
Comment (bgruening) [inline chemicaltoolbox/simsearch/simsearch.xml:5]:
```suggestion
        <token name="@GALAXY_VERSION@">1</token>
```

File: chemicaltoolbox/xchem-deep/transfs.xml [hunk 12]
```diff
@@ -1,4 +1,4 @@
-<tool id="xchem_transfs_scoring" name="XChem TransFS pose scoring" version="0.4.0">
+<tool id="xchem_transfs_scoring" name="XChem TransFS pose scoring" version="1.3+galaxy0">
```
Comment (bgruening) [inline chemicaltoolbox/xchem-deep/transfs.xml:1]:
```suggestion
<tool id="xchem_transfs_scoring" name="XChem TransFS pose scoring" version="1.3+galaxy0" profile="23.0">
```

**Review by bgruening — COMMENTED** [review]
There are some very nice changes in it! Thanks a lot.

Can you please add a profile version to all tools relevant? And then commit that with the other changes.
Afterwards I propose to rename the .shed.yml tool for the pgtools and the fragmenter, if they don't have tests they should not be pushed to the TS. So skip them for now.

File: chemicaltoolbox/fragmenter/merger.xml (hunk 4 above)
Comment (hechth) [reply chemicaltoolbox/fragmenter/merger.xml:33]: Well the planemo lint complains if there is no shed file if I am not mistaken.

**Comment by hechth** [comment]
@bgruening maybe I'll try to split this PR to the openbabel tools which I actually need and the rest, though I am afraid they are entangled through the documentation and the links.

File: chemicaltoolbox/fragmenter/merger.xml (hunk 4 above)
Comment (bgruening) [reply chemicaltoolbox/fragmenter/merger.xml:33]: Yes, do that as very last step. I need to push then manually the good tools.

File: chemicaltoolbox/openmg/openmg.xml [hunk 13]
```diff
@@ -1,9 +1,9 @@
 <tool id="ctb_openmg" name="Open Molecule Generator" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
```
Comment (bgruening) [inline chemicaltoolbox/openmg/openmg.xml:1]:
```suggestion
<tool id="ctb_openmg" name="Open Molecule Generator" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@" profile="23.0">
```

File: chemicaltoolbox/fragmenter/merger.xml [hunk 14]
```diff
@@ -1,12 +1,11 @@
-<tool id="bgchem_fragment_merger" name="Merging" version="0.1">
+<tool id="bgchem_fragment_merger" name="Merging" version="0.1+galaxy0">
```
Comment (bgruening) [inline chemicaltoolbox/fragmenter/merger.xml:1]:
```suggestion
<tool id="bgchem_fragment_merger" name="Merging" version="0.1+galaxy0" profile="23.0">
```

File: chemicaltoolbox/chemfp/mol2fps.xml [hunk 15]
```diff
@@ -1,13 +1,13 @@
 <tool id="ctb_chemfp_mol2fps" name="Molecule to fingerprint" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@">
```
Comment (bgruening) [inline chemicaltoolbox/chemfp/mol2fps.xml:1]:
```suggestion
<tool id="ctb_chemfp_mol2fps" name="Molecule to fingerprint" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
```

**Pull request merged**